File: buildSrc/src/main/java/com/doctor/aistring/StringEncryptor.java
package com.doctor.aistring;

import org.gradle.api.logging.Logger;
import org.gradle.api.logging.Logging;

import java.io.Serializable;
import java.nio.charset.StandardCharsets;
import java.util.Base64;

import javax.crypto.Cipher;
import javax.crypto.spec.SecretKeySpec;

public class StringEncryptor implements Serializable {
    private transient final String algorithm = "AES/ECB/PKCS5Padding";
    private static final Logger logger = Logging.getLogger(StringEncryptor.class);

    public byte[] encrypt(String input, String key) {
        try {
            // 将十六进制密钥转换为字节数组
            byte[] keyBytes = new byte[32];
            for (int i = 0; i < key.length(); i += 2) {
                keyBytes[i / 2] = (byte) Integer.parseInt(key.substring(i, i + 2), 16);
            }
            SecretKeySpec keySpec = new SecretKeySpec(keyBytes, algorithm);

            Cipher cipher = Cipher.getInstance(algorithm);
            cipher.init(Cipher.ENCRYPT_MODE, keySpec);

            // 使用 UTF-8 编码
            return cipher.doFinal(input.getBytes(StandardCharsets.UTF_8));
        } catch (Exception e) {
            // 如果AES加密失败，回退到简单的XOR加密
            byte[] inputBytes = input.getBytes(StandardCharsets.UTF_8);
            byte[] result = new byte[inputBytes.length];
            int keyHash = key.hashCode();
            for (int i = 0; i < inputBytes.length; i++) {
                result[i] = (byte) (inputBytes[i] ^ keyHash);
            }
            return result;
        }
    }
}

File: buildSrc/src/main/kotlin/com/doctor/aistring/StringEncryptor.kt
package com.doctor.aistring

import javax.crypto.Cipher
import javax.crypto.spec.SecretKeySpec
import java.security.MessageDigest
import java.util.Base64
import java.io.Serializable
import org.gradle.api.logging.Logger
import org.gradle.api.logging.Logging

class StringEncryptor : Serializable {
    @Transient
    private val algorithm = "AES/ECB/PKCS5Padding"
    @Transient
    private val logger: Logger = Logging.getLogger(StringEncryptor::class.java)
    
    fun encrypt(input: String, key: String): ByteArray {
        try {
            // 将十六进制密钥转换为字节数组
            val keyBytes = key.chunked(2).map { it.toInt(16).toByte() }.toByteArray()
            val keySpec = SecretKeySpec(keyBytes, "AES")
            
            val cipher = Cipher.getInstance(algorithm)
            cipher.init(Cipher.ENCRYPT_MODE, keySpec)
            
            logger.lifecycle("使用密钥: $key")
            logger.lifecycle("密钥字节数组: ${keyBytes.joinToString(",")}")
            
            // 使用 UTF-8 编码
            val result = cipher.doFinal(input.toByteArray(Charsets.UTF_8))
            logger.lifecycle("加密结果: ${result.joinToString(",")}")
            return result
        } catch (e: Exception) {
            logger.error("AES加密失败，使用XOR加密", e)
            // 如果AES加密失败，回退到简单的XOR加密
            return input.toByteArray(Charsets.UTF_8).map { 
                it.toInt() xor key.hashCode() 
            }.map { it.toByte() }.toByteArray()
        }
    }
    
    private fun generateKey(key: String): SecretKeySpec {
        val digest = MessageDigest.getInstance("SHA-256")
        val bytes = key.toByteArray(Charsets.UTF_8)
        val hash = digest.digest(bytes)
        return SecretKeySpec(hash.copyOf(16), "AES")
    }
}